connectors/redis/redigo.go

```diff
@@ -88,6 +90,11 @@ func (c *simpleRedis) Shutdown() error {
 // Do is a proxy method that calls Redigo's `Do` method and returns its output. It remembers
 // to close connections taken from the pool
 func (c *simpleRedis) do(commandName string, args ...interface{}) (interface{}, error) {
+	if c.stats != nil {
```

Comment:
Let's write a `NoOp` impelmentation of the stats interface so we don't have to do all these nil checks.

Reply:
i'll get this in first and then add a noop separately and change all the nil checks at once